Core objects

jrosgiralt opened the pull request.

jrosgiralt committed:
Automatically generate a dootrip title; Added src and dst address field to the dootrip contenty type

Commit:
Fixed typographical errors according to the code checker

jrosgiralt committed:
Added departure and arrival dates to dootrip content type

jrosgiralt committed:
Set non-admin user theme to Gin

jrosgiralt committed:
Created Labdoo Blocks Module; implemented initial block for hubs and edoovillages

jrosgiralt committed:
Fixed coding style nits

jrosgiralt committed:
Simple updates to the edoovillage/hub block

greggmarshall commented:
Jordi, there are a few automated checks to clean up first
```
gitpod@labdoo3-web:/var/www/html$ phpcs --standard="Drupal,DrupalPractice" --extensions=php,module,inc,install,test,profile,theme,css,info,txt,md,yml web/modules/custom

FILE: /var/www/html/web/modules/custom/lbd_default_content/lbd_default_content.module
-------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------
 7 | ERROR | [x] Whitespace found at end of line
-------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------


FILE: /var/www/html/web/modules/custom/lbd_blocks/src/Plugin/Block/BlockHubEdoovillage.php
---------------------------------------------------------------------------------------------------------------------------------------------
FOUND 7 ERRORS AND 4 WARNINGS AFFECTING 10 LINES
---------------------------------------------------------------------------------------------------------------------------------------------
 19 | ERROR   | [x] Opening brace should be on the same line as the declaration
 23 | ERROR   | [x] Expected 1 blank line before function; 0 found
 24 | ERROR   | [x] Opening brace should be on the same line as the declaration
 28 | WARNING | [ ] t() calls should be avoided in classes, use \Drupal\Core\StringTranslation\StringTranslationTrait and $this->t() instead
 28 | WARNING | [ ] Only string literals should be passed to t() where possible
 33 | WARNING | [ ] Only string literals should be passed to t() where possible
 41 | ERROR   | [x] Opening brace should be on the same line as the declaration
 49 | ERROR   | [x] Opening brace should be on the same line as the declaration
 50 | WARNING | [ ] Unused variable $config.
 59 | ERROR   | [x] Opening brace should be on the same line as the declaration
 61 | ERROR   | [x] Expected 1 blank line after function; 0 found
---------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 7 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------------------------------------------

Time: 174ms; Memory: 6MB
```
and
```
gitpod@labdoo3-web:/var/www/html$ phpstan --level=5 analyze web/modules/custom
Note: Using configuration file /var/www/html/phpstan.neon.
 3/3 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------- 
  Line   lbd_blocks/src/Plugin/Block/BlockHubEdoovillage.php  
 ------ ----------------------------------------------------- 
  26     Undefined variable: $code                            
 ------ ----------------------------------------------------- 


                                                                                                                        
 [ERROR] Found 1 error                                                                                                  
                     
```

Commit:
Fixed format errors/warnings

jrosgiralt committed:
Fixed format errors/warnings

jrosgiralt commented:
Sorry i had run the checks in the immediately before commit, but must have missed doing it in the last commit. I was able to resolve all of the errors/warnings except for one warning:
```
gitpod@labdoo3-web:/var/www/html$ ./vendor/bin/phpcs --standard="Drupal,DrupalPractice" --extensions=php,module,inc,install,test,profile,theme,css,info,txt,md,yml web/modules/custom

FILE: /var/www/html/web/modules/custom/lbd_blocks/src/Plugin/Block/BlockHubEdoovillage.php
------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
------------------------------------------------------------------------------------------
 34 | WARNING | Only string literals should be passed to t() where possible
------------------------------------------------------------------------------------------

Time: 154ms; Memory: 6MB
```

The offending line is this one:

```
    return [
      '#markup' => $this->t($code),
    ];
```

The following fixes the warning, however it does not render the html markup correctly:

```
    $replacements['@code'] = $code;
    return [
      '#markup' => $this->t("@code", $replacements),
    ];
```

greggmarshall commented:
You shouldn't put variables in the t function.  Take a look at https://www.drupal.org/docs/7/api/localization-api/dynamic-strings-with-placeholders.  It would be best to translate the strings before wrapping them with HTML.

jrosgiralt commented:
Thanks Gregg. I know we should not put variables in the t function. The correct way should be the second codeblock I gave you, where I use the replacements array. But doing so prevents the html markup from properly rendering on the webpage.

greggmarshall commented:
Did you try the other placeholders in that documentation???

And again, it is better to translate the string(s) then wrap them in HTML.

jrosgiralt commented:
Try with exclamation point instead of the “@“
But the right thing to do it to put the ts up there in the code and let them be out of the html code.

jrosgiralt committed:
Created bootstrap_labdoo subtheme from bootstrap_barrio and enabled it by default

jrosgiralt committed:
Updated docs

jrosgiralt committed:
Added Labdoo images

jrosgiralt committed:
Show hub/edoovillage block only in hub/edoovillage content types; hide title

jrosgiralt committed:
- Added Labdoo 2.0 roles
- Removed content editor role

jrosgiralt committed:
- Added paragraph module
- Created Labdoo Story content type using paragraph module
- Created Gallery content type
- Added logic to Plugin/Block/BlockHubEdoovillage.php to check if user has the correct right to access/edit an edoovillage/hub.

jrosgiralt committed:
added module drupal/entity_reference_revisions
